src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -509,11 +509,7 @@ void SpatialPooler::compute(SDR &input, bool learn, SDR &active) {
   calculateOverlap_(input, overlaps_);
   calculateOverlapPct_(overlaps_, overlapsPct_);
 
-  if (learn) {
-    boostOverlaps_(overlaps_, boostedOverlaps_);
-  } else {
-    boostedOverlaps_.assign(overlaps_.begin(), overlaps_.end());
```

Comment:
The microsoft compiler goes crazy when the assign( ) must perform a type conversion and it would result in possible-loss-of-data.  It gives a page of text, but in the end it is only a warning.  Is there a way to do this such that the compiler knows it is ok?

Reply:
Did you try:
`nupic::utils::VectorHelpers::castVectorType<T1, T2>( vector<T1> ) -> vector<T2>`

Reply:
> nupic::utils::VectorHelpers::castVectorType<T1, T2>( vector<T1> ) -> vector<T2>

Oh, forgot about those routines.
I will give it a try.  Thanks.